packages/iota/src/validation/outputs/outputs.ts

```diff
@@ -22,13 +29,43 @@ import { validateNftOutput } from "./nftOutput";
  */
 export function validateOutputs(outputs: OutputTypes[], protocolInfo: INodeInfoProtocol): IValidationResult {
     const results: IValidationResult[] = [];
+    let totalAmount: number = 0;
+    let nativeTokenCount: number = 0;
+
+    if (outputs.length < MIN_OUTPUT_COUNT || outputs.length > MAX_OUTPUT_COUNT) {
+        results.push({
+            isValid: false,
+            errors: [`Outputs count must be between ${MIN_OUTPUT_COUNT} and ${MAX_OUTPUT_COUNT}.`]
+        });
+    }
 
     for (const output of outputs) {
+        totalAmount += Number(output.amount);
+        if (output.type !== TREASURY_OUTPUT_TYPE && output.nativeTokens) {
+            const tokenIds = output.nativeTokens.map(token => token.id);
```

Comment:
This sounds like it belongs inside `validateOutput`, or even lower in validate native tokens (if it's not there already).